File: LeetCode/Java/876-ordered-map-链表的中间结点.java
/*
 * @lc app=leetcode.cn id=876 lang=java
 *
 * [876] 链表的中间结点
 */

// @lc code=start
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode res = head, p = head;
        int count = 0;
        while (p != null) {
            ++count;
            p = p.next;
            if (count % 2 == 0)
                res = res.next;
        }
        return res;
    }
}
// @lc code=end

File: LeetCode/Go/876-ordered-map-链表的中间结点.go
/*
 * @lc app=leetcode.cn id=876 lang=golang
 *
 * [876] 链表的中间结点
 */
package leetcode

// @lc code=start
func middleNode(head *ListNode) *ListNode {
	var slow, fast = head, head
	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next.Next
	}
	return slow
}

// @lc code=end
